src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js

```diff
@@ -537,22 +537,24 @@ export default class StandardRouteModel extends BaseModel {
      *
      * @for StandardRouteModel
      * @method _updateWaypointsWithPreviousWaypointData
-     * @parma waypointModelList {array<StandardRouteWaypointModel>}
+     * @param waypointModelList {array<StandardRouteWaypointModel>}
      * @private
      */
     _updateWaypointsWithPreviousWaypointData(waypointModelList) {
-        _forEach(waypointModelList, (waypoint, i) => {
-            let previousWaypoint = waypointModelList[i - 1];
-            if (i === 0) {
-                previousWaypoint = waypoint;
+        for (let i = 1; i < waypointModelList.length; i++) {
```

Comment:
We can only do useful math when we have two waypoints (hence why the previous had `if (i === 0) { nextWaypoint = waypoint; }`). So we start at the second waypoint where we have our first opportunity to calculate a distance. Removal of the lodash was in favor of the native for loop for the ability to start at i=1.